### ISSUE TITLE
Cannot read properties of undefined (reading 'component') while loading Elemental

### ISSUE THREAD

**Issue by eva-vashkevich** [rancher/dashboard#11414]
**Setup**
- Rancher version: 2.9-head
- Rancher UI Extensions: latest Elemental master
- Browser type & version: Chrome

**Describe the bug**

TypeError: Cannot read properties of undefined (reading 'component')
    at VueComponent.emberLink (index.vue:231:1)
    at push../node_modules/vue/dist/vue.runtime.esm.js.Watcher.get (vue.runtime.esm.js:3446:1)
    at push../node_modules/vue/dist/vue.runtime.esm.js.Watcher.evaluate (vue.runtime.esm.js:3547:1)
    at VueComponent.computedGetter [as emberLink] (vue.runtime.esm.js:5537:1)
    at Object.get (vue.runtime.esm.js:5329:1)
    at Proxy.render (index.vue:4:1)
    at Vue._render (vue.runtime.esm.js:2684:1)
    at VueComponent.updateComponent (vue.runtime.esm.js:3864:1)
    at push../node_modules/vue/dist/vue.runtime.esm.js.Watcher.get (vue.runtime.esm.js:3446:1)
    at push../node_modules/vue/dist/vue.runtime.esm.js.Watcher.run (vue.runtime.esm.js:3522:1)
_callee$ @ client.js:105

error is thrown in the console

**To Reproduce**

1. Add the elemental-ui repo to `local cluster` -> `apps` -> `repositories`:
type `git`
url: `https://github.com/rancher/elemental-ui`
branch: `gh-pages`
2. Install Elemental version `1.3.1-rc7` (only published version compliant with Rancher 2.9)
3. Start elemental locally (install extension and operator + crds)
4. Go to `cluster management` -> hit `create` -> toggle the rke indicator to rke1
5. Go to Create Elemental cluster (via Elemental extension -> dashboard -> button `Create Elemental Cluster`

**Result**
"Cannot read properties of undefined (reading 'component')" error is shown

**Expected Result**

Cluster creation wizard should load

**Screenshots**

<img width="1720" alt="Screenshot 2024-07-11 at 3 00 00 PM" src="https://github.com/rancher/dashboard/assets/133266076/ca7b26ec-7c4f-4903-bade-aaebe6ac3617">

**Additional context**

**Comment by richard-cox** [comment]
Think this comes from 

```
  // if driver type has a custom form component, don't load an ember page
          if (this.selectedSubType.component) {
            return '';
          }
          // For RKE1 and hosted Kubernetes Clusters, set the ember link
          // so that we load the page rather than using RKE2 create
          if (this.selectedSubType?.emberLink) {
            return this.selectedSubType.emberLink;
          }
```

The fix might be as simple as `selectedSubType?.component`. @mantis-toboggan-md & @aalves08 can you co-ordinate and test?

**Comment by aalves08** [comment]
Tried that fix and it's a rabbit hole. 

I’ve finally found the problem:
If this is toggled to `rke1` (check, then it doesn’t find the correct machine config  :exploding_head:
It’s persistent as well. Fix is on Elemental side, just adding to the create cluster url `rkeType=rke2`  like `https://<-BASE-URL->/c/_/manager/provisioning.cattle.io.cluster/create?type=machineinventoryselectortemplate&rkeType=rke2`

<img width="2321" alt="Screenshot 2024-07-11 at 15 30 02" src="https://github.com/rancher/dashboard/assets/97888974/afa19c5b-b75d-4020-82a6-4ea9bef8c0ae">

**Comment by aalves08** [comment]
@eva-vashkevich @richard-cox updated repro steps on issue.

Also I don't think this is a release-blocker. There's a workaround (have the toggle in rke2 position in `cluster management` -> `create` screen). From my tests this will only affect Elemental. Unfortunately, the correct fix is not on Elemental side. Will put up a PR

**Comment by aalves08** [comment]
Due to the area that this is in and since it needs a lot of moving parts, I deemed the issue as `QA/manual-test`

**Comment by yonasberhe23** [comment]
QA blocked by https://github.com/rancher/elemental-ui/issues/202

**Comment by aalves08** [comment]
@yonasberhe23 I am really sorry (I forgot about adding crucial information for you to test this...).

I've updated the repro steps both in the issue + PR now. You will need version `1.3.1-rc7` of Elemental to test this (from `elemental-ui` repo) since that's the only published version of Elemental that works with Rancher 2.9 right now.

You still might hit https://github.com/rancher/elemental-ui/issues/202 though, but at least now you'll have a **working** version of Elemental to be tested in 2.9. 🙏

**Comment by yonasberhe23** [comment]
> [@yonasberhe23](https://github.com/yonasberhe23) I am really sorry (I forgot about adding crucial information for you to test this...).
> 
> I've updated the repro steps both in the issue + PR now. You will need version `1.3.1-rc7` of Elemental to test this (from `elemental-ui` repo) since that's the only published version of Elemental that works with Rancher 2.9 right now.
> 
> You still might hit [rancher/elemental-ui#202](https://github.com/rancher/elemental-ui/issues/202) though, but at least now you'll have a **working** version of Elemental to be tested in 2.9. 🙏
> 
> PR to fix the above issue has just been merged [#11448](https://github.com/rancher/dashboard/pull/11448)

No worries @aalves08 ! thanks for the info. I'll give this a test now

**Comment by yonasberhe23** [comment]
Tested in 
- v2.9-81337b95660cba868629d6fe769bd7e8242b5aee-head
- master 3da1205

Interface loads properly using the above repro steps

https://github.com/user-attachments/assets/79118246-c0fb-48a1-b0e0-a88eedc54975
